servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RequestTracker.java

```diff
@@ -39,17 +51,12 @@ interface RequestTracker {
      */
     void onSuccess(long beforeStartTimeNs);
 
-    /**
-     * Records cancellation of the action for which latency is to be tracked.
-     *
-     * @param beforeStartTimeNs return value from {@link #beforeStart()}.
-     */
-    void onCancel(long beforeStartTimeNs);
-
     /**
      * Records a failed completion of the action for which latency is to be tracked.
      *
      * @param beforeStartTimeNs return value from {@link #beforeStart()}.
+     * @param errorClass the class of error that triggered this method.
+     * @param cause the specific cause of the error, if available. Primarily for observability.
      */
-    void onError(long beforeStartTimeNs);
+    void onError(long beforeStartTimeNs, ErrorClass errorClass, @Nullable Throwable cause);
```

Comment:
I agree the `cause` is useless in here, not even for logging, it becomes too verbose. I would argue that we start without it.

Reply:
Deal. We can always add it back later if we want.